app/lib/submission_builder/state_file.rb

```diff
@@ -0,0 +1,19 @@
+module SubmissionBuilder
+  module StateFile
+    def from_state_code_and_year(state_code, year)
+      unless ::StateFile::StateInformationService::STATES_INFO.key?(state_code)
+        raise StandardError state_code
+      end
+      builder = "SubmissionBuilder::Ty#{year.to_i}::States::#{state_code.to_s.titleize}::IndividualReturn"
+      builder.constantize
+    end
+
+    def from_state_code(state_code)
+      # TODO: I am hard coding 2022 as the current year here because we don't have anything else defined yet
+      from_state_code_and_year(state_code, 2022)
+    end
+
+    module_function :from_state_code_and_year
+    module_function :from_state_code
```

Comment:
i tried to look up `module_function` but it wasn't immediately clear what this is doing, could you explain briefly?

Reply:
This allows the function to be invoked from the module rather than including it in a class and invoking it from there.